encoder_config.go

```diff
@@ -0,0 +1,153 @@
+	// used for serialization.
+	EnableName bool `json:"enableName" yaml:"enableName"`
+
+	// EnableStacktrace controls if a stacktrace should be serialized when available.
+	EnableStacktrace bool `json:"enableStacktrace" yaml:"enableStacktrace"`
+
+	// EnableCaller controls if the entry caller should be serialized.
+	// If enabled, the EncodeCaller configuration is used for serialization.
+	EnableCaller bool `json:"enableCaller" yaml:"enableCaller"`
+
+	// LineEnding defines the string used for line endings.
+	LineEnding string `json:"lineEnding" yaml:"lineEnding"`
+
+	// EncodeName defines how to encode a loggers name.
+	// It will only be applied if EnableName is set to true.
+	EncodeName zapcore.NameEncoder `json:"nameEncoder" yaml:"nameEncoder"`
+
+	// EncodeLevel sets the log level for which any context should be logged.
+	EncodeLevel zapcore.LevelEncoder `json:"levelEncoder" yaml:"levelEncoder"`
+
+	// EncodeDuration sets the format for encoding time.Duration values.
+	EncodeDuration zapcore.DurationEncoder `json:"durationEncoder" yaml:"durationEncoder"`
+
+	// EncodeCaller defines how an entry caller should be serialized.
+	// It will only be applied if EnableCaller is set to true.
+	EncodeCaller CallerEncoder `json:"callerEncoder" yaml:"callerEncoder"`
+}
+
+// NewDefaultEncoderConfig returns an EncoderConfig with default settings.
+func NewDefaultEncoderConfig() EncoderConfig {
+	return EncoderConfig{
+		EnableName:       true,
+		EnableCaller:     true,
+		EnableStacktrace: true,
+		LineEnding:       defaultLineEnding,
+		EncodeName:       defaultEncodeName,
+		EncodeLevel:      defaultEncodeLevel,
+		EncodeDuration:   defaultEncodeDuration,
+		EncodeCaller:     defaultEncodeCaller,
+	}
+}
+
+// ECSCompatibleEncoderConfig takes an existing zapcore.EncoderConfig
+// and sets ECS relevant configuration options to ECS conformant values.
+// The returned zapcore.EncoderConfig can be used to create
+// an ECS conformant encoder.
+// Be aware that this will always replace any set EncodeCaller function
+// with the ecszap.ShortCallerEncoder.
+// This is a pure convenience function for making a transition from
+// existing an zap logger to an ECS conformant zap loggers easier.
+// It is recommended to make use of the ecszap.EncoderConfig whenever possible.
+func ECSCompatibleEncoderConfig(cfg zapcore.EncoderConfig) zapcore.EncoderConfig {
+	cfg.MessageKey = messageKey
+	cfg.LevelKey = loglevelKey
+	cfg.TimeKey = timeKey
+	cfg.EncodeTime = timeEncoder
+	if cfg.EncodeDuration == nil {
+		cfg.EncodeDuration = zapcore.NanosDurationEncoder
+	}
+	if cfg.NameKey != "" {
+		cfg.NameKey = "log.logger"
+	}
+	if cfg.StacktraceKey != "" {
+		cfg.StacktraceKey = "log.origin.stacktrace"
+	}
+	if cfg.CallerKey != "" {
+		cfg.CallerKey = "log.origin"
+		cfg.EncodeCaller = defaultEncodeCaller
+	}
+	if cfg.EncodeLevel == nil {
+		cfg.EncodeLevel = defaultEncodeLevel
+	}
+	return cfg
+}
+
+func toZapCoreEncoderConfig(cfg EncoderConfig) zapcore.EncoderConfig {
```

Comment:
Perhaps make this a method of `EncoderConfig`, and export it? I think it could be nice for the "advanced use-case" example where we want to wrap an underlying Core with an ECS encoder, but using the narrower ecszap.EncoderConfig.

e.g.

```go
encoderConfig := ecszap.EncoderConfig{...}
encoder := zapcore.NewJSONEncoder(encoderConfig.ToZapcore())
syslogCore := newSyslogCore(encoder, level)
core := ecszap.WrapCore(syslogCore)
```